libsrc/osic1p/crt0.s

```diff
@@ -0,0 +1,54 @@
+; ---------------------------------------------------------------------------
+; crt0.s
+; ---------------------------------------------------------------------------
+;
+; Startup code for Ohio Scientific Challenger 1P
+
+.export   _init, _exit
+.import   _main
+
+.export   __STARTUP__ : absolute = 1    ; Mark as startup
+.import   __RAM_START__, __RAM_SIZE__   ; Linker generated
+
+.import    zerobss, initlib, donelib
+
+.include  "zeropage.inc"
+.include  "extzp.inc"
+.include  "osic1p.inc"
+
+; ---------------------------------------------------------------------------
+; Place the startup code in a special segment
+
+.segment  "STARTUP"
+
+; ---------------------------------------------------------------------------
+; A little light 6502 housekeeping
+
+_init:    ldx     #$FF          ; Initialize stack pointer to $01FF
+          txs
+          cld                   ; Clear decimal mode
+
+; ---------------------------------------------------------------------------
+; Set cc65 argument stack pointer
+
+          lda     #<(__RAM_START__ + __RAM_SIZE__)
+          sta     sp
+          lda     #>(__RAM_START__ + __RAM_SIZE__)
```

Comment:
I'm surprised that we didn't see this bug before now!  That expression points to the bottom of the C data stack.  It needs to point to the top, that is: "(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)".